Purchases/Purchasing/TrialOrIntroPriceEligibilityChecker.swift

```diff
@@ -92,6 +93,23 @@ class TrialOrIntroPriceEligibilityChecker {
         return introDict
     }
 
+    @available(iOS 11.2, macOS 10.13.2, macCatalyst 13.0, tvOS 11.2, watchOS 6.2, *)
+    func productsWithIntroOffers(productIdentifiers: [String], completion: @escaping ReceiveIntroEligibilityBlock) {
+        self.productsManager.products(withIdentifiers: Set(productIdentifiers)) { products in
+            let eligibility: [(String, IntroEligibility)] = Array(products).compactMap {
+                guard $0.introductoryPrice != nil else {
+                    return nil
+                }
+                return ($0.productIdentifier, IntroEligibility(eligibilityStatus: .eligible))
```

Comment:
FFTI: I had a bit of trouble understanding this block at first. Would this be easier to understand at a glance? 

```swift
let eligibility: [(String, IntroEligibility)] =
    Array(products)
    .filter { $0.introductoryPrice != nil }
    .map { ($0.productIdentifier, IntroEligibility(eligibilityStatus: .eligible)) }

let productIdsToIntroEligibleStatus = Dictionary(uniqueKeysWithValues: eligibility)
```

Reply:
So, I like what I wrote because, technically, we are compacting. I also agree with you, your solution is more readable. I think that's more important here 😄.

Reply:
Updated ✅